### PR TITLE
scansaphana: SAP HANA scanning implementation

### PR DESCRIPTION
**Pull request by vinzenz**
scansaphana: SAP HANA scanning implementation

    This patch introduces detection of SAP HANA and adds a check for it.
    It is assumed that SAP Hana instances are installed to /hana/shared.

    From there leapp will detect all instances located there and parses the
    manifest file to retrieve information about the installed binaries.
    Additionally it will perform a check to determine whether or not SAP
    Hana has currently any active instances. If the check later determines
    that there are any running instances the upgrade will be inhibited, as
    upgrading with running SAP Hana instances is not supported.

    The tests for discovery will also recreate as much as is needed to fake
    the environment to make the code think that SAP Hana is either not
    installed/running or that there are several instances.
    It will also check that it will detect all instances that are faked.

### CODE REVIEW

**Comment by pirat89** [comment]
@vinzenz It's really weird but....in case the sap is not running, regarding the current implementation, how we retrieve the information about the install sap version-release? With the shutdown of the whole sap service, regarding the current implementation, it's possible to do upgrade with sap hana unsupported on rhel 8. As well, the sap hana has to be detected on the system to be able to apply other needed actions (e.g. just e4s repositories are enabled, different upgrade path, ...)

File: repos/system_upgrade/el7toel8/actors/ipuworkflowconfig/libraries/ipuworkflowconfig.py [hunk 1]
```diff
@@ -60,5 +65,36 @@ def get_booted_kernel():
...
+        for entry in os.listdir(HANA_BASE_PATH):
+            # Does /hana/shared/{entry}/exe/linuxx86_64/hdb/sapcontrol exist?
+            if os.path.exists(os.path.join(HANA_BASE_PATH, entry, HANA_SAPCONTROL_PATH)):
+                return True
+    return False
+
+
+def detect_sap_hana():
+    """ Detect SAP Hana either by existance of saphostctrl or sapcontrol """
+    if os.path.exists(HANA_SAPHOSTCTRL_PATH):
```
Comment (vinzenz) [inline repos/system_upgrade/el7toel8/actors/ipuworkflowconfig/libraries/ipuworkflowconfig.py:92]: if this detects all SAP stuff this and the line after need to go away.

So basically it would consist only of the call to detect_hana_shared_sapcontrol, that would detect SAP Hana only.

File: repos/system_upgrade/el7toel8/actors/checksaphana/libraries/checksaphana.py [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+
+
+def running_check(info):
+    if info.running:
+        reporting.create_report([
+            reporting.Title('Found running SAP HANA instances'),
+            reporting.Summary(
+                'In order to perform a system upgrade it is necessary that all instances of SAP Hana are stopped.'
+                'Please shutdown all SAP Hana instances before you continue with the upgrade.'
+            ),
```
Comment (pirat89) [inline repos/system_upgrade/el7toel8/actors/checksaphana/libraries/checksaphana.py:23]: the instruction should be part of remediation hint. no big problem but i guess it would be better if we put this stuff into the remediation as hint always. Some people would go for remediations mainly during reading.

File: repos/system_upgrade/el7toel8/actors/checksaphana/libraries/checksaphana.py [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+    if found:
+        detected = _create_detected_instances_list(found)
+        reporting.create_report([
+            reporting.Title('Found outdated SAP HANA 1'),
+            reporting.Summary(
+                ('SAP Hana 1.00 is not supported on the version of RHEL you are'
+                 ' upgrading to. In order to upgrade RHEL, you will have to'
+                 ' upgrade SAP Hana first to the latest version of SAP Hana 2.00.\n\n'
+                 'The following instances have been detected to be running version 1.00:\n'
+                 '{}'.format(detected))
```
Comment (pirat89) [inline repos/system_upgrade/el7toel8/actors/checksaphana/libraries/checksaphana.py:69]: nitpick. do we have a doc, ..., so we could attach the link for info about supported version of sap?

File: repos/system_upgrade/el7toel8/actors/ipuworkflowconfig/libraries/ipuworkflowconfig.py (hunk 1 above)
Comment (pirat89) [reply repos/system_upgrade/el7toel8/actors/ipuworkflowconfig/libraries/ipuworkflowconfig.py]: you are right. let's do that.

File: repos/system_upgrade/el7toel8/actors/checksaphana/actor.py [hunk 4]
```diff
@@ -0,0 +1,19 @@
+from leapp.actors import Actor
+from leapp.libraries.actor.checksaphana import perform_check
+from leapp.models import SapHanaInfo
+from leapp.reporting import Report
+from leapp.tags import FactsPhaseTag, IPUWorkflowTag
```
Comment (pirat89) [inline repos/system_upgrade/el7toel8/actors/checksaphana/actor.py:5]: Checksphase?

File: repos/system_upgrade/el7toel8/actors/ipuworkflowconfig/libraries/ipuworkflowconfig.py (hunk 1 above)
Comment (vinzenz) [reply repos/system_upgrade/el7toel8/actors/ipuworkflowconfig/libraries/ipuworkflowconfig.py]: This part has been removed to ensure we only detect SAP Hana

File: repos/system_upgrade/el7toel8/actors/scansaphana/libraries/scansaphana.py [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+        output = run([
+            'sudo', '-u', admin_name, sapcontrol_path, '-nr', instance_number, '-function', 'GetProcessList'])
+        if output['exit_code'] == 3:
+            # GetProcessList succeeded, all processes running correctly
+            return True
+        if output['exit_code'] == 4:
+            # GetProcessList succeeded, all processes stopped
+            return False
+        # SAP Hana might be somewhere in between (Starting/Stopping)
+        return len(output['stdout'].split('\n')) > 7
```
Comment (pirat89) [inline repos/system_upgrade/el7toel8/actors/scansaphana/libraries/scansaphana.py:91]: @vinzenz  any typical output what could be here during stopping? or a comment why detection "less than 8 lines" signifies the status for sure? regarding the possibility the services is starting, would be better to retry to get the result a time later? I understand that as it's possible the running sap hana instance could be marked as stopped even when it's starting actually without any warning.

File: repos/system_upgrade/el7toel8/actors/ipuworkflowconfig/libraries/ipuworkflowconfig.py [hunk 6]
```diff
@@ -60,5 +68,29 @@ def get_booted_kernel():
         )
 
 
-def get_target_version():
-    return os.getenv('LEAPP_DEVEL_TARGET_RELEASE', None) or CURRENT_TARGET_VERSION
+def get_target_version(flavour=LEAPP_UPGRADE_FLAVOUR_DEFAULT):
+    """
+    Returns the target version for the given `flavour` of upgrade. The default value for `flavour` is `default`.
+    In case the environment variable `LEAPP_DEVEL_TARGET_RELEASE` is set, the value of it will be returned.
+    """
```
Comment (pirat89) [inline repos/system_upgrade/el7toel8/actors/ipuworkflowconfig/libraries/ipuworkflowconfig.py:75]:
```suggestion
    """
    Returns the target version for the given `flavour` of upgrade. The default value for `flavour` is `default`.

    In case the environment variable `LEAPP_DEVEL_TARGET_RELEASE` is set, the value of it will be returned.
    """
```

File: repos/system_upgrade/el7toel8/actors/checksaphana/actor.py (hunk 4 above)
Comment (vinzenz) [reply repos/system_upgrade/el7toel8/actors/checksaphana/actor.py:5]: duh ofc

File: repos/system_upgrade/el7toel8/models/saphanainfo.py [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+    topic = SystemInfoTopic
+
+    name = fields.String()
+    """ Name of the instance """
+    path = fields.String()
+    """ Directory of the SAP HANA instance """
+    instance_number = fields.String()
+    """ SAP HANA Instance number """
+    running = fields.Boolean()
+    """ Is the instance is currently running? """
```
Comment (pirat89) [inline repos/system_upgrade/el7toel8/models/saphanainfo.py:16]:
```suggestion
    """ Is the instance currently running? """
```

File: repos/system_upgrade/el7toel8/models/saphanainfo.py [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+    admin = fields.String()
+    """ Name of the instance administrator """
+    manifest = fields.JSON()
```
Comment (pirat89) [inline repos/system_upgrade/el7toel8/models/saphanainfo.py:19]: @vinzenz  I guess you used the JSON because we are not sure what all values could be in the manifest? Otherwise it could be represented e.g. by SapHanaManifest model with the specified fields. We are strict about the use of JSON type. let's sync. but I prefer to replace it if we know what could be inside and it's not tooo long list.

File: repos/system_upgrade/el7toel8/actors/scansaphana/libraries/scansaphana.py (hunk 5 above)
Comment (vinzenz) [reply repos/system_upgrade/el7toel8/actors/scansaphana/libraries/scansaphana.py:91]: @pirat89 
Running

```hxeadm@hxehost:/usr/sap/HXE/HDB90> ./exe/sapcontrol -nr 90 -function GetProcessList

26.01.2021 17:55:17
GetProcessList
OK
name, description, dispstatus, textstatus, starttime, elapsedtime, pid
hdbdaemon, HDB Daemon, GREEN, Running, 2020 05 08 21:56:28, 6307:58:49, 16043
hdbcompileserver, HDB Compileserver, GREEN, Running, 2020 05 08 21:57:00, 6307:58:17, 16365
hdbdiserver, HDB Deployment Infrastructure Server, GREEN, Running, 2020 05 08 21:57:37, 6307:57:40, 16546
hdbindexserver, HDB Indexserver-HXE, GREEN, Running, 2020 05 08 21:57:01, 6307:58:16, 16391
hdbnameserver, HDB Nameserver, GREEN, Running, 2020 05 08 21:56:29, 6307:58:48, 16061
hdbwebdispatcher, HDB Web Dispatcher, GREEN, Running, 2020 05 08 21:57:38, 6307:57:39, 16549
```

Stopping
hxeadm@hxehost:/usr/sap/HXE/HDB90> ./exe/sapcontrol -nr 90 -function GetProcessList

26.01.2021 17:56:26
GetProcessList
OK
name, description, dispstatus, textstatus, starttime, elapsedtime, pid
hdbdaemon, HDB Daemon, YELLOW, Stopping, 2020 05 08 21:56:28, 6307:59:58, 16043
hdbcompileserver, HDB Compileserver, YELLOW, Scheduled, , , -1
hdbdiserver, HDB Deployment Infrastructure Server, YELLOW, Scheduled, , , -1
hdbindexserver, HDB Indexserver-HXE, YELLOW, Scheduled, , , -1
hdbnameserver, HDB Nameserver, YELLOW, Stopping, 2020 05 08 21:56:29, 6307:59:57, 16061
hdbwebdispatcher, HDB Web Dispatcher, YELLOW, Scheduled, , , -1
```
Stopped
```
hxeadm@hxehost:/usr/sap/HXE/HDB90> ./exe/sapcontrol -nr 90 -function GetProcessList

26.01.2021 17:56:29
GetProcessList
OK
name, description, dispstatus, textstatus, starttime, elapsedtime, pid
hdbdaemon, HDB Daemon, GRAY, Stopped, , , 16043

Comment (pirat89) [reply repos/system_upgrade/el7toel8/actors/scansaphana/libraries/scansaphana.py:91]:
```suggestion
        # in case of the Starting/Stopping there is always more than 7 lines
        return len(output['stdout'].split('\n')) > 7
```

File: utils/docker-tests/Dockerfile [hunk 9]
```diff
@@ -3,13 +3,12 @@ FROM centos:7
...
 RUN yum update -y && \
-    yum install python-virtualenv make git -y
+    yum install python-virtualenv python-setuptools make git -y
 
 # NOTE(ivasilev) Unless we upgrade pip an obsolete v 9 will be on the system
 # and we need at least 10.0.1 to install a not-yet-merged framework pr in case
 # it's necessary
-RUN curl "https://bootstrap.pypa.io/get-pip.py" -o "get-pip.py" && \
-    python get-pip.py
+RUN easy_install pip
```
Comment (Rezney) [inline utils/docker-tests/Dockerfile:11]: what is the reason for this change?

File: repos/system_upgrade/el7toel8/actors/scansaphana/tests/test_scansaphana.py [hunk 10]
```diff
@@ -0,0 +1,177 @@
...
+        cur_path.mkdir()
+
+    linkpath = base_path.join(manifest_path_parts[0])
+
+    manifest = cur_path.join(manifest_path_parts[-1])
+    manifest.write_text(unicode_type(SAPHANA2_MANIFEST), encoding='utf-8')
+
+    sapcontrol = cur_path.join(scansaphana.HANA_SAPCONTROL_PATH.split('/')[-1])
+    sapcontrol.write_text(unicode_type('''#!/bin/bash
+echo 'Line1'
```
Comment (Rezney) [inline repos/system_upgrade/el7toel8/actors/scansaphana/tests/test_scansaphana.py:142]: could you please elaborate on what we are trying to achieve with this test?

File: repos/system_upgrade/el7toel8/actors/scansaphana/libraries/scansaphana.py [hunk 11]
```diff
@@ -0,0 +1,95 @@
...
+    except CalledProcessError:
+        api.current_logger().warn(
+            'Failed to retrieve SAP HANA instance status from sapcontrol - Considering it not running')
```
Comment (Rezney) [inline repos/system_upgrade/el7toel8/actors/scansaphana/libraries/scansaphana.py:94]:
```suggestion
            'Failed to retrieve SAP HANA instance status from sapcontrol - Considering it is not running')
```

File: repos/system_upgrade/el7toel8/actors/checksaphana/libraries/checksaphana.py [hunk 12]
```diff
@@ -0,0 +1,171 @@
...
+
+def perform_check():
+    """ Performs all checks for SAP Hana and will skip if the upgrade flavour is not `saphana` """
+
+    if api.current_actor().configuration.flavour != 'saphana':
+        # Do not run on non saphana upgrades
+        return
+
+    if not platform_check():
+        # If this architecture is not supported, there's no sense in continuation.
```
Comment (Rezney) [inline repos/system_upgrade/el7toel8/actors/checksaphana/libraries/checksaphana.py:162]:
```suggestion
        # If this architecture is not supported, there's no sense in continuing.
```

File: repos/system_upgrade/el7toel8/actors/checksaphana/libraries/checksaphana.py [hunk 13]
```diff
@@ -0,0 +1,171 @@
...
+
+
+def platform_check():
+    """ Creates an inhibitor report in case the system is not running on x86_64 """
+    if not architecture.matches_architecture(architecture.ARCH_X86_64):
+        reporting.create_report([
+            reporting.Title('SAP Hana upgrades are only supported on X86_64 systems'),
+            reporting.Summary(
+                ('Upgrades for SAP Hana are only supported on X86_64 systems.'
+                 ' For more information please regard to the documentation.')
```
Comment (Rezney) [inline repos/system_upgrade/el7toel8/actors/checksaphana/libraries/checksaphana.py:139]:
```suggestion
                 ' For more information please consult the documentation.')
```

File: repos/system_upgrade/el7toel8/actors/checksaphana/libraries/checksaphana.py [hunk 14]
```diff
@@ -0,0 +1,171 @@
...
+            continue
+        if not _fullfills_hana_min_version(instance):
+            _add_hana_details(found, instance)
+
+    if found:
+        detected = _create_detected_instances_list(found)
+        reporting.create_report([
+            reporting.Title('SAP Hana needs to be updated before upgrade'),
+            reporting.Summary(
+                ('Please update SAP Hana to at least {min_hana_version} to continue the upgrade.\n\n'
```
Comment (Rezney) [inline repos/system_upgrade/el7toel8/actors/checksaphana/libraries/checksaphana.py:120]:
```suggestion
                ('Please update SAP Hana to at least {min_hana_version} to continue with the upgrade.\n\n'
```

File: repos/system_upgrade/el7toel8/actors/checksaphana/actor.py [hunk 15]
```diff
@@ -0,0 +1,28 @@
...
+
+
+class CheckSapHana(Actor):
+    """
+    If SAP Hana has been detected, several checks are performed to ensure a successful upgrade.
+
+    If the upgrade flavour is 'default' no checks are being executed.
+
+    The following checks are executed:
+    - If this is system is _NOT_ running x86_64 the upgrade is inhibited
```
Comment (Rezney) [inline repos/system_upgrade/el7toel8/actors/checksaphana/actor.py:15]:
```suggestion
    - If this system is _NOT_ running on x86_64, the upgrade is inhibited
```

**Review by Rezney — COMMENTED** [review]
Overall looks good to me, just left some minor suggestions/question...

File: repos/system_upgrade/el7toel8/models/saphanainfo.py (hunk 8 above)
Comment (Rezney) [reply repos/system_upgrade/el7toel8/models/saphanainfo.py:19]: some manifest examples are in one of the tests, not sure if it is worth making it a model...

File: repos/system_upgrade/el7toel8/actors/scansaphana/libraries/scansaphana.py (hunk 5 above)
Comment (Rezney) [reply repos/system_upgrade/el7toel8/actors/scansaphana/libraries/scansaphana.py:91]: What about putting these examples to the code?

File: repos/system_upgrade/el7toel8/actors/setuptargetrepos/actor.py [hunk 16]
```diff
@@ -59,7 +77,8 @@ def process(self):
                 for repo in repo_file.data:
                     enabled_repos.add(repo.repoid)
 
-        skipped_repos = skipped_repos.intersection(set(used.keys()))
+        # TODO(pstodulk): the part of the workaround mentioned above..
+        skipped_repos = skipped_repos.intersection(set(used.keys())) - EXTRAS_REPOS
```
Comment (Rezney) [inline repos/system_upgrade/el7toel8/actors/setuptargetrepos/actor.py:81]: @pirat89 we have here "mapped_repos", "used_repos", "rhel_repos", and "skipped_repos", I think it deserves some commentary :) It can wait but I guess potential reviewers can benefit from it in the future...

Comment (pirat89) [reply repos/system_upgrade/el7toel8/actors/setuptargetrepos/actor.py:81]: Yeah. I spend a time checking all "tentacles" :) This actor is one of those that are expected to be rewritten in future when there is a time :) after the `targetuserspacecreator` is rewritten first.

**Comment by pirat89** [comment]
@vinzenz btw, rebase against the master to make travis passing again.

File: repos/system_upgrade/el7toel8/actors/scansaphana/libraries/scansaphana.py [hunk 17]
```diff
@@ -0,0 +1,98 @@
...
+        return s
+
+    data = {}
+    try:
+        with open(path, 'r') as f:
+            for line in _decoded(f.read()).split('\n'):
+                try:
+                    key, value = line.split(':', 1)
+                except ValueError:
+                    continue
```
Comment (pirat89) [inline repos/system_upgrade/el7toel8/actors/scansaphana/libraries/scansaphana.py:33]: when this happens and why the continue is ok? a comment will be more clear.

File: utils/docker-tests/Dockerfile (hunk 9 above)
Comment (pirat89) [reply utils/docker-tests/Dockerfile:11]: @Rezney  https://github.com/oamg/leapp-repository/pull/637 - rebase will drop this commit as it's already present in master.

File: repos/system_upgrade/el7toel8/models/saphanainfo.py (hunk 8 above)
Comment (pirat89) [reply repos/system_upgrade/el7toel8/models/saphanainfo.py:19]: @Rezney  we are relatively strongly against the JSON models and we require people to write the correct models instead. I prefer to not have an exception if not needed. This one seems to be easy to cover by model.

Comment (pirat89) [reply repos/system_upgrade/el7toel8/models/saphanainfo.py:19]: But if we do now know whole manifest - so it's problematic to cover correctly in all possible outputs - it's ok to keep json, at least for now.

**Comment by Rezney** [comment]
Wondering whether we should not create also an info report that customers are on SAP HANA upgrade path. I see we have report for running instance, outdated instance, instance update needed, only x86_64 supported. Maybe they have some forgotten SAP instance but they want to upgrade it as a non-SAP? Does it make sense? @pirat89 wdyt?

File: repos/system_upgrade/el7toel8/actors/scansaphana/libraries/scansaphana.py (hunk 17 above)
Comment (vinzenz) [reply repos/system_upgrade/el7toel8/actors/scansaphana/libraries/scansaphana.py:44]: Comment added

File: repos/system_upgrade/el7toel8/actors/scansaphana/libraries/scansaphana.py (hunk 11 above)
Comment (vinzenz) [reply repos/system_upgrade/el7toel8/actors/scansaphana/libraries/scansaphana.py]: That one changed actually to 'Considering it as not running'  which makes the `is` weird.

File: repos/system_upgrade/el7toel8/actors/scansaphana/tests/test_scansaphana.py (hunk 10 above)
Comment (vinzenz) [reply repos/system_upgrade/el7toel8/actors/scansaphana/tests/test_scansaphana.py:143]: This was just a fake output. I can replace it with real output examples.

File: repos/system_upgrade/el7toel8/models/saphanainfo.py (hunk 8 above)
Comment (vinzenz) [reply repos/system_upgrade/el7toel8/models/saphanainfo.py:19]: There are changing values in the manifest, and I wanted to keep it completely for debugging purposes.
We do not know the whole manifest and there are keys that can change from release to release.
e.g. 
```
max_rel2.0_sps0_rev-number: 02
max_rel2.0_sps0_patchlevel: 02
max_rel2.0_sps1_rev-number: 12
max_rel2.0_sps1_patchlevel: 05
max_rel2.0_sps2_rev-number: 24
max_rel2.0_sps2_patchlevel: 10
max_rel2.0_sps3_rev-number: 37
max_rel2.0_sps3_patchlevel: 07
max_rel2.0_sps4_rev-number: 48
max_rel2.0_sps4_patchlevel: 02
```

I change change it for something more strict.

File: repos/system_upgrade/el7toel8/actors/checksaphana/libraries/checksaphana.py (hunk 3 above)
Comment (berndfinger) [reply repos/system_upgrade/el7toel8/actors/checksaphana/libraries/checksaphana.py:77]: https://launchpad.support.sap.com/#/notes/2235581

File: repos/system_upgrade/el7toel8/actors/checksaphana/libraries/checksaphana.py [hunk 18]
```diff
@@ -0,0 +1,213 @@
...
+
+
+def platform_check():
+    """ Creates an inhibitor report in case the system is not running on x86_64 """
+    if not architecture.matches_architecture(architecture.ARCH_X86_64):
+        reporting.create_report([
+            reporting.Title('SAP HANA upgrades are only supported on X86_64 systems'),
+            reporting.Summary(
+                ('Upgrades for SAP HANA are only supported on X86_64 systems.'
+                 ' For more information please consult the documentation.')
```
Comment (berndfinger) [inline repos/system_upgrade/el7toel8/actors/checksaphana/libraries/checksaphana.py:181]: The supported hardware architectures are also contained in https://launchpad.support.sap.com/#/notes/2235581.

File: repos/system_upgrade/el7toel8/actors/checksaphana/libraries/checksaphana.py (hunk 3 above)
Comment (pirat89) [reply repos/system_upgrade/el7toel8/actors/checksaphana/libraries/checksaphana.py:77]: I am not able to access the content. And ideally we should point just to redhat official documentation - not outside doc..

**Review by pirat89 — APPROVED** [review]
the code lgtm.

**Comment by pirat89** [comment]
> Wondering whether we should not create also an info report that customers are on SAP HANA upgrade path. I see we have report for running instance, outdated instance, instance update needed, only x86_64 supported. Maybe they have some forgotten SAP instance but they want to upgrade it as a non-SAP? Does it make sense? @pirat89 wdyt?

I have had a similar question and I guess that's for @berndfinger and @shaded-enmity to decide that, as it's more bussiness question than a technical one. However I am more for the version that when they have installed sap, they are going to the the ugprade in the SAP way. They can always ask for support exception - e.g. to remove all SAP related actors, use devel opts, etc. I guess we could find just couple of users that would like to do it as it seems more like very corner case. Definitely, if people want it, we will get know via Pavel :-D

**Comment by vinzenz** [comment]
> > Wondering whether we should not create also an info report that customers are on SAP HANA upgrade path. I see we have report for running instance, outdated instance, instance update needed, only x86_64 supported. Maybe they have some forgotten SAP instance but they want to upgrade it as a non-SAP? Does it make sense? @pirat89 wdyt?
> 
> @Rezney I have had a similar question and I guess that's for @berndfinger and @shaded-enmity to decide that, as it's more bussiness question than a technical one. However I am more for the version that when they have installed sap, they are going to the the ugprade in the SAP way. They can always ask for support exception - e.g. to remove all SAP related actors, use devel opts, etc. I guess we could find just couple of users that would like to do it as it seems more like very corner case. Definitely, if people want it, we will get know via Pavel :-D

They could do that by either temporarily unmounting /hana/shared or renaming it

**Comment by pirat89** [comment]
> They could do that by either temporarily unmounting /hana/shared or renaming it

So no problem at all let's say.

**Review by pirat89 — APPROVED** [review]
re-approve :)

**Comment by AloisMahdal** [comment]
As of *leapp-repository-0.12.0-0.202102021241Z.65993c6.saphana_initial.PR503.el7_8*. I can confirm that **murphy-ci/integration** has passed most of tests and can be waived.  (Just few test cases were missed because of timeout.)

**Review by berndfinger — COMMENTED** [review]
Tested (preupgrade as well as upgrade) of a RHEL 7.7 system running SAP HANA (latest available 1.0 as well as various revisions of 2.0), either running or stopped.
In case SAP HANA is installed using installation path /hana/shared, leapp correctly detects the HANA version and inhibits a RHEL 7.7 to 8.2 upgrade of a SAP HANA version which is not supported on RHEL 7.7 or on RHEL 8.2. In all other cases, it allows an upgrade and correctly upgrades to 8.2, enables the correct e4s repos, and correctly locks the RHEL minor release to 8.2
Also the messages on the terminal and in /var/log/leapp/leapp-report.txt are looking good to me.
So this is a PASS from my side.

**Comment by AloisMahdal** [comment]
We've checked the **continuous-integration/oamg-ci/e2e** with @pirat89 and @vinzenz and there's some curl write error while downloading the RHEL-8 content. @pirat89 says it's a known error and not related to leapp (also @Rezney should already know about it) so it's OK to waive.
